tidy3d/components/heat_charge/time_spec.py

```diff
@@ -0,0 +1,73 @@
+"""Dealing with time specifications for DeviceSimulation"""
+
+from abc import ABC
+from typing import Optional, Union
+
+import pydantic.v1 as pd
+
+from ...constants import KELVIN, SECOND, VOLT
+from ..base import Tidy3dBaseModel
+
+
+class TimeBaseSpec(Tidy3dBaseModel, ABC):
+    """Base class to deal with time specifications for DeviceSimulations"""
+
+
+class SteadySpec(TimeBaseSpec):
+    """Defines a steady-state specification
+
+    Example
+    --------
+    >>> import tidy3d as td
+    >>> time_spec = td.SteadySpec()
+    """
+
+
+class UnsteadySpec(TimeBaseSpec):
+    """Defines an unsteady specification
+
+    Example
+    --------
+    >>> import tidy3d as td
+    >>> time_spec = td.UnsteadySpec(
+    ...     time_step=0.01,
+    ...     total_time_steps=200,
+    ...     initial_value=300,
+    ...     output_fr=50,
+    ... )
+    """
+
+    time_step: pd.PositiveFloat = pd.Field(
+        title="Time-step",
+        description="Time step taken for each iteration of the time integration loop.",
+        units=SECOND,
+    )
+
+    total_time_steps: pd.PositiveInt = pd.Field(
+        title="Total time steps",
+        description="Specifies the total number of time steps run during the simulation.",
+    )
+
+    initial_temperature: Optional[pd.PositiveFloat] = pd.Field(
+        None,
+        title="Initial temperature.",
+        description="Initial value for the temperature field.",
+        units=KELVIN,
+    )
+
+    initial_voltage: Optional[float] = pd.Field(
+        None,
+        title="Initial voltage.",
+        description="Initial value for the voltage field.",
+        units=VOLT,
+    )
+
+    output_fr: Optional[pd.PositiveInt] = pd.Field(
+        default=1,
+        title="Output frequency",
+        description="Determines how often output files will be written. I.e., an output "
+        "file will be written every 'output_fr' time steps.",
+    )
+
+
+TimeSpecType = Union[SteadySpec, UnsteadySpec]
```

Comment:
Wonder if this needs to be abstracted for compatibility with RF to some level in terms of what is steady and nonsteady in a given simulation and making sure the data types are the same? Just some ideas not sure of the implementation in progress on this

Reply:
Like we probably want to come up with a well-defined standard way of these steady-temporal relationships because they have a strong effect on multiphysical simulations ultimately. Some things are static when some things are dynamic and it gets represented by this type of implementation strategies that we want to make sure are unified in order to more flow control

Reply:
hmmm steady/unsteady in heat is well defined. Either way, this needs to come out of this PR :sweat_smile:

Reply:
Haha all good!